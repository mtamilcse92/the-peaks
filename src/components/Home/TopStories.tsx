import React from 'react';
import Image from '../common/Image';
import Spinner from '../common/Spinner';
import Dropdown from '../common/Dropdown';
import Bookmark from '../common/Bookmark';
import ContentCard from '../common/ContentCard';
import { useStories } from '../../hooks/stories'

{/* <Image imageURL="https://www.industrialempathy.com/img/remote/ZiClJf-1280w.avif" /> */}
const Home = () => {
    const { data } = useStories()
    
    return (
      <div className="App">
        <Image />
        <Spinner isVisible />
        <Dropdown />
        <Bookmark label="ADD BOOKMARK" onClick={() => {}} />
        <ContentCard description="Hellooo..." title="Test title client" subtitle="helloooo" />
      </div>
    );
}

export default Home;
