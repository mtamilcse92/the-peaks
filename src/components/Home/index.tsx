import React, { Fragment } from "react";
import { useHistory } from "react-router-dom";
import Spinner from "../common/Spinner";
import Dropdown from "../common/Dropdown";
import Bookmark from "../common/Bookmark";
import PageHeader from "../common/PageHeader";
import ContentCard from "../common/ContentCard";
import { useStories } from "../../hooks/stories";
import { paths } from '../../constants/router';
import { PersistContext } from '../../context/persistContext';
import styles from "./styles.module.scss";

const Home = () => {
  const history = useHistory();
  const {state: persistState} = React.useContext(PersistContext)
  const [sortBy, setSortBy] = React.useState<string>('relevance')
  const q = persistState.searchText || ''
  const query = q ? { q, 'order-by': sortBy } : { 'order-by': sortBy }
  const { data, refetch, isFetching, isLoading } = useStories(query);
  const showLoader = isFetching || isLoading;

  React.useEffect(() => {
    refetch()
  }, [refetch, q, sortBy])

  return (
    <section className={styles.container}>
      <PageHeader
        title={(q || '').length !== 0 ? "Search Result" : "Top stories"}
        rhsElement={
          <Fragment>
            <Bookmark label="VIEW BOOKMARK" onClick={() => history.push(paths.bookmarks)} />
            <Dropdown onSelect={(item) => setSortBy(item.value)} />
          </Fragment>
        }
      />
      {showLoader && <Spinner isVisible />}
      {!showLoader && (
        <div className={styles.contentWrapper}>
          {data?.map((item) => (
            <div key={item.id} className={styles.content}>
              <ContentCard
                onClick={() => history.push(`${paths.contentDetail}?id=${item.id}`)}
                title={item.webTitle}
                imageUrl={item?.fields?.thumbnail}
                subtitle={item?.fields?.headline}
              />
            </div>
          ))}
        </div>
      )}
    </section>
  );
};

export default Home;
