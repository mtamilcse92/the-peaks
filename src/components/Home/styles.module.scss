@import '../../styles/index.scss';
@import '../../styles/mixin.scss';
@import '../../styles/variables.scss';

.container {
    display: flex;
    align-items: center;
    flex-direction: column;
}

.contentWrapper {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
}

.content {
    display: flex;
    height: 400px;
    width: 350px;
    margin: 10px;
    @include media(tablet) {
        width: 312px;
    }
}
