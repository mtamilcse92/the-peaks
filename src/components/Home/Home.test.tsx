import React from "react";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Home from "./index";
import PersistContextProvider from '../../context/persistContext';
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import "@testing-library/jest-dom";
const queryClient = new QueryClient();

describe("<Home  /> spec", () => {
  it("renders the Home", () => {
    const history = createMemoryHistory();
    history.push(
      `/?id=film/2021/mar/29/selfie-review-gallic-slaves-to-the-algorithm`
    );
    const Component = render(
      <PersistContextProvider>
      <QueryClientProvider client={queryClient}>
        <Router history={history}>
          <Home />
        </Router>
      </QueryClientProvider>
      </PersistContextProvider>
    );
    expect(Component).toMatchSnapshot();
  });
});

