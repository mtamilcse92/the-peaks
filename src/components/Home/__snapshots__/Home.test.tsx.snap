// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home  /> spec renders the Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container"
      >
        <header
          class="header"
        >
          <span
            class="heading"
          >
            Top stories
          </span>
          <div
            class="rhs"
          >
            <button
              class="container"
            >
              <svg
                class="icon"
              >
                bookmark_24px.svg
              </svg>
              <span
                class="label"
              >
                VIEW BOOKMARK
              </span>
            </button>
            <div
              class="dropdown"
            >
              <div
                class="dropdownHeader"
              >
                Select your destination
                <svg>
                  arrow_drop_down_24px.svg
                </svg>
              </div>
              <div
                class="dropdownBody"
              >
                <div
                  class="dropdownItem"
                  data-item="[object Object]"
                  id="1"
                >
                  Newest first
                </div>
                <div
                  class="dropdownItem"
                  data-item="[object Object]"
                  id="2"
                >
                  Oldest first
                </div>
                <div
                  class="dropdownItem"
                  data-item="[object Object]"
                  id="3"
                >
                  Most popular
                </div>
              </div>
            </div>
          </div>
        </header>
        <div
          class="container"
        >
          <div
            class="spinner"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container"
    >
      <header
        class="header"
      >
        <span
          class="heading"
        >
          Top stories
        </span>
        <div
          class="rhs"
        >
          <button
            class="container"
          >
            <svg
              class="icon"
            >
              bookmark_24px.svg
            </svg>
            <span
              class="label"
            >
              VIEW BOOKMARK
            </span>
          </button>
          <div
            class="dropdown"
          >
            <div
              class="dropdownHeader"
            >
              Select your destination
              <svg>
                arrow_drop_down_24px.svg
              </svg>
            </div>
            <div
              class="dropdownBody"
            >
              <div
                class="dropdownItem"
                data-item="[object Object]"
                id="1"
              >
                Newest first
              </div>
              <div
                class="dropdownItem"
                data-item="[object Object]"
                id="2"
              >
                Oldest first
              </div>
              <div
                class="dropdownItem"
                data-item="[object Object]"
                id="3"
              >
                Most popular
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="container"
      >
        <div
          class="spinner"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
